src/borg/repository.py

```diff
@@ -960,12 +960,11 @@ def replay_segments(self, index_transaction_id, segments_transaction_id):
             self.rollback()
 
     def get_storage_quota_use(self):
-        storage_quota_use = 0
-        for i, (segment, filename) in enumerate(self.io.segment_iterator()):
-            objects = list(self.io.iter_objects(segment))
-            for tag, key, offset, size, _ in objects:
-                storage_quota_use += header_size(tag) + size
-        return storage_quota_use
+        hints_path = os.path.join(self.path, "hints.%d" % self.get_transaction_id())
+        integrity_data = self._read_integrity(self.get_transaction_id(), "hints")
+        with IntegrityCheckedFile(hints_path, write=False, integrity_data=integrity_data) as fd:
+            hints = msgpack.unpack(fd)
+            return hints["storage_quota_use"]
```

Comment:
also, guess this function is way too specific.

there is a lot of useful data in the hints file, not just the storage_quota_use, so I guess we should either:
- always load (or even better: demand load) the hints file and make it available via repository object attributes (like it is already done when starting a transaction), but make sure it is only saved when it needs to be saved.
- make a function _load_hints that returns the full hints object and use it from everywhere where the hints need loading.

Reply:
I guess I can add a getter to the variables in the hints that checks if the file has been loaded and load it.
Something like this:
```python3
@property
def shadow_index(self):
    if self._shadow_index = {}:
        hints = self._load_hints()
        # Initialize every variable
    return self.shadow_index()
```
This way, the hints file is only loaded when needed, but that method would basically be repeated for all the variables in the hints.

Reply:
All related attributes should be `None` until initialised.

Reply:
I can't change the init values of `shadow_index` and `storage_quota_use` to None. I'm gonna use `info` to load the hints file.
So, getting the used quota is done like this: `storage_quota=format_file_size(repository.info().get("storage_quota_use"))`
And the info methods looks like this:
``` python3
def info(self):
        """return some infos about the repo (must be opened first)"""
        self._load_hints()
        return dict(
            id=self.id, version=self.version, append_only=self.append_only, storage_quota_use=self.storage_quota_use
        )
```